tests/c/x64_imm_add.c

```diff
@@ -0,0 +1,52 @@
+// ignore-if: test $(uname -m) != x86_64
```

Comment:
I don't think this should be a C test: it should be a unit test. C tests are slow and this one doesn't do anything a unit test couldn't AFAICS.

Reply:
We have unit tests in addition to this. If you are happy for other (less targeted) tests to check the runtime semantics of the optimisation, we can delete this one though.

Entirely up to you.

Reply:
What I'm saying is I *think* everything this C test tests for *should* be covered in unit tests in ykrt *and* those tests will run much, much, quicker. If I'm right about those things, I'd prefer to delete this test.